src/main/java/graphql/nadel/engine/ExecutionResultNodeMapper.java

```diff
@@ -0,0 +1,91 @@
+
+        Map<String, String> typeRenameMappings = environment.typeRenameMappings;
+        GraphQLSchema overallSchema = environment.overallSchema;
+
+        // maybe we need that later
+//        GraphQLOutputType mappedType = mapFieldType(executionStepInfo, typeRenameMappings, overallSchema);
+        ExecutionPath mappedPath = pathMapper.mapPath(node.getExecutionPath(), node.getResultKey(), environment);
+
+        GraphQLObjectType mappedObjectType = mapObjectType(node, typeRenameMappings, overallSchema);
+        GraphQLFieldDefinition mappedFieldDefinition = getFieldDef(overallSchema, mappedObjectType, node.getFieldName());
+
+        return node.transform(builder -> builder
+                .executionPath(mappedPath)
+                .objectType(mappedObjectType)
+                .fieldDefinition(mappedFieldDefinition)
+        );
+
+    }
+
+    private GraphQLObjectType mapObjectType(ExecutionResultNode node, Map<String, String> typeRenameMappings, GraphQLSchema overallSchema) {
+        String objectTypeName = mapTypeName(typeRenameMappings, node.getObjectType().getName());
+        GraphQLObjectType mappedObjectType = overallSchema.getObjectType(objectTypeName);
+        assertNotNull(mappedObjectType, "object type %s not found in overall schema", objectTypeName);
+        return mappedObjectType;
+    }
+
+//    private GraphQLOutputType mapFieldType(ExecutionStepInfo executionStepInfo, Map<String, String> typeRenameMappings, GraphQLSchema overallSchema) {
+//        return mapFieldType(executionStepInfo.getType(), overallSchema, typeRenameMappings);
+//    }
+
+    private String mapTypeName(Map<String, String> typeRenameMappings, String name) {
+        return typeRenameMappings.getOrDefault(name, name);
+    }
+
+//    private GraphQLOutputType mapFieldType(GraphQLOutputType graphQLOutputType,
+//                                           GraphQLSchema overallSchema,
+//                                           Map<String, String> typeRenameMappings) {
+//        if (isNotWrapped(graphQLOutputType)) {
+//            String typeName = mapTypeName(typeRenameMappings, ((GraphQLNamedOutputType) graphQLOutputType).getName());
+//            GraphQLOutputType outputType = (GraphQLOutputType) overallSchema.getType(typeName);
+//            return assertNotNull(outputType, "type %s not found in overall schema for field type", ((GraphQLNamedOutputType) graphQLOutputType).getName());
+//        }
+//        if (isList(graphQLOutputType)) {
+//            return list(mapFieldType((GraphQLOutputType) ((GraphQLList) graphQLOutputType).getWrappedType(), overallSchema, typeRenameMappings));
+//        }
+//        if (isNonNull(graphQLOutputType)) {
+//            return nonNull(mapFieldType((GraphQLOutputType) ((GraphQLNonNull) graphQLOutputType).getWrappedType(), overallSchema, typeRenameMappings));
+//        }
+//        return Assert.assertShouldNeverHappen();
+//    }
```

Comment:
remove dead code or leave it uncalled - but dont leave it commented out

Reply:
done